arch/risc-v/src/common/espressif/esp_wifi_utils.c

```diff
@@ -0,0 +1,643 @@
+              iwe->cmd = SIOCGIWENCODE;
+              iwe->u.data.flags = IW_ENCODE_ENABLED | IW_ENCODE_NOKEY;
+              iwe->u.data.length = 0;
+              iwe->u.essid.pointer = NULL;
+
+              priv->scan_result_size += ESP_IW_EVENT_SIZE(data);
+              result_size -= ESP_IW_EVENT_SIZE(data);
+
+              /* Copy AP channel */
+
+              if (result_size < ESP_IW_EVENT_SIZE(freq))
+                {
+                  goto scan_result_full;
+                }
+
+              iwe = (struct iw_event *)
+                    &priv->scan_result[priv->scan_result_size];
+              iwe->len = ESP_IW_EVENT_SIZE(freq);
+              iwe->cmd = SIOCGIWFREQ;
+              iwe->u.freq.e = 0;
+              iwe->u.freq.m = ap_list_buffer[bss_count].primary;
+
+              priv->scan_result_size += ESP_IW_EVENT_SIZE(freq);
+              result_size -= ESP_IW_EVENT_SIZE(freq);
+            }
+        }
+
+      parse_done = true;
+    }
+
+scan_result_full:
+
+  /* Continue instead of break to log dropped AP results */
+
+  if (parse_done == false)
+    {
+      wlerr("ERROR: No more space in scan_result buffer\n");
+    }
+
+  if (ap_list_buffer)
+    {
+      kmm_free(ap_list_buffer);
+      ap_list_buffer = NULL;
+    }
+
+  priv->scan_status = ESP_SCAN_DONE;
+  nxsem_post(&priv->scan_signal);
+}
+
+/****************************************************************************
+ * Name: esp_wifi_to_errno
+ *
+ * Description:
+ *   Transform from ESP Wi-Fi error code to NuttX error code
+ *
+ * Input Parameters:
+ *   ret - ESP Wi-Fi error code
+ *
+ * Returned Value:
+ *   NuttX error code defined in errno.h
+ *
+ ****************************************************************************/
+
+int32_t esp_wifi_to_errno(int ret)
+{
+  int err;
+
+  /* Unmask component error bits */
+
+  err = ret & 0xfff;
+
+  if (ret < ESP_ERR_WIFI_BASE)
+    {
+      if (err == ESP_OK)
+        {
+          return OK;
+        }
+      else
+        {
+          wlerr("ERROR: %s\n", esp_err_to_name(ret));
+          if (err == ESP_ERR_NO_MEM)
+            {
+              return -ENOMEM;
+            }
+          else if (err == ESP_ERR_INVALID_ARG)
+            {
+              return -EINVAL;
+            }
+          else if (err == ESP_ERR_INVALID_STATE)
+            {
+              return -EIO;
+            }
+          else if (err == ESP_ERR_INVALID_SIZE)
+            {
+              return -EINVAL;
+            }
+          else if (err == ESP_ERR_NOT_FOUND)
+            {
+              return -ENOSYS;
+            }
+          else if (err == ESP_ERR_NOT_SUPPORTED)
+            {
+              return -ENOSYS;
+            }
+          else if (err == ESP_ERR_TIMEOUT)
+            {
+              return -ETIMEDOUT;
+            }
+          else if (err == ESP_ERR_INVALID_MAC)
+            {
+              return -EINVAL;
+            }
+          else
+            {
+              return ERROR;
+            }
```

Comment:
```suggestion
```

Reply:
I didn't get the proposed changes... even copying and pasting it doesn't seem to contain any differences.

Reply:
The suggestion is to remove part of the code as duplication of a line 642

Reply:
Oh, yeah. I just refactored it, can you please take a look?